core/templates/pages/contributor-dashboard-page/translation-language-selector/translation-language-selector.component.spec.ts

```diff
@@ -258,4 +262,33 @@ describe('Translation language selector', () => {
         selectedLanguage);
     });
   });
+
+  it('should toggle dropdown', () => {
+    component.ngOnInit();
+
+    expect(component.dropdownShown).toBe(false);
+
+    component.toggleDropdown();
+
+    expect(component.dropdownShown).toBe(true);
+    expect(component.optionsFilter).toBe('');
+    expect(component.filteredOptions).toBe(component.options);
```

Comment:
Please add a pre-check for this.

Reply:
Added a pre-check in commit bb229ef638db198e72708c956f15d37c9c5a3f3b, however I am not 100% sure that it is exactly what you meant with `pre-check`.

Reply:
The `component.filteredOptions` doesn't seem to change after the `toggleDropdown` function is called.
Ideally, we want to check if a change occurs in the value. In this case, we have to test if `this.filteredOptions = this.options;` in line 110 is working as expected.